### PR TITLE
faddat/upgrade go

### PR DESCRIPTION
**Pull request by faddat**
This PR upgrades the go relayer to go v1.21 in advance of upgrading to sdk 50.  It is meant to reduce the size of a PR that fully upgrades to 50.

- upgrade ci to go 1.21
- upgrade to go1.21

### CODE REVIEW

**Comment by faddat** [comment]
is "TESTING - interchaintest / scenarios (pull_request) Failing after 11m" 

known to be flaky at all?  I figure this out to be good to go

**Comment by jtieri** [comment]
> is "TESTING - interchaintest / scenarios (pull_request) Failing after 11m"
> 
> known to be flaky at all? I figure this out to be good to go

yes unfortunately our self hosted runner encounters oom errors. we have been discussing moving over to GitHub's hosted runners and making sure that each test case runs in its own job to avoid this issue.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -15,7 +15,7 @@ jobs:
       - name: Set up Go 1.20
```
Comment (jtieri) [inline .github/workflows/build.yml]: can you also adjust the name to accurately reference the version of Go that is being used? better yet we could change this line to simply be `Setting up Go` that way we don't have to keep making changes every time we update the Go version being used.

File: .github/workflows/interchaintest.yml [hunk 2]
```diff
@@ -13,7 +13,7 @@ jobs:
       - name: Set up Go 1.20
```
Comment (jtieri) [inline .github/workflows/interchaintest.yml]: same as my last comment, maybe we quit referencing a specific Go version in the names that way we don't need to keep making changes every time we update the Go version.

**Review by jtieri — CHANGES_REQUESTED** [review]
just the comment regarding the Go version being referenced in the workflow files then this should be good to merge, thanks!

**Review by jtieri — APPROVED** [review]

**Pull request merged**